ts/modules/full-screen.src.ts

```diff
@@ -18,63 +15,189 @@ import H from '../parts/Globals.js';
+};
+
+// Helping function - to close fullscreen, use toggleFullscreen() instead.
+H.Chart.prototype.closeFullscreen = function (this: Highcharts.Chart): void {
+    var chart = this,
+        exportingOptions = chart.options.exporting,
+        exportDivElements = chart.exportDivElements,
+        menuItems = exportingOptions?.buttons?.contextButton.menuItems;
+
+    // Don't fire exitFullscreen() when user exited using 'Escape' button.
+    if (
+        chart.isFullscreen &&
+        chart.container.ownerDocument instanceof Document
+    ) {
+        chart.container.ownerDocument[chart.browserProps.exitFullscreen]();
+    }
+
+    // Replace button text.
+    if (
+        exportDivElements &&
+        exportDivElements.length &&
+        exportingOptions?.menuItemDefinitions?.viewFullscreen &&
+        menuItems &&
+        chart.options.lang?.viewFullscreen
+    ) {
+        exportDivElements[menuItems.indexOf('viewFullscreen')].innerHTML = (
+            exportingOptions.menuItemDefinitions.viewFullscreen.text ||
+            chart.options.lang.viewFullscreen
+        );
+    }
+
+    // Unbind event as it's necessary only before exiting from fullscreen.
+    if (chart.unbindFullscreenEvent) {
+        chart.unbindFullscreenEvent();
+    }
+
+    chart.isFullscreen = false;
+};
+
 /* eslint-disable no-invalid-this, valid-jsdoc */
 
 /**
- * The FullScreen class.
- * The module allows user to enable full screen mode in StockTools.
- * Based on default solutions in browsers.
+ * Exporting module required. Toggles displaying the chart in fullscreen mode.
+ * By default, when the exporting module is enabled, a context button with
+ * a drop down menu in the upper right corner accesses this function.
  *
- * @private
- * @class
- * @name Highcharts.FullScreen
+ * @since       next
  *
- * @param {Highcharts.HTMLDOMElement} container
- *        Chart container
+ * @sample      highcharts/members/chart-togglefullscreen/
+ *              Toggle fullscreen mode from a HTML button
+ *
+ * @function    Highcharts.Chart#toggleFullscreen
+ * @return      {void}
+ * @requires    modules/exporting
+ * @requires    modules/fullscreen
  */
-var FullScreen = H.FullScreen = function (
-    this: Highcharts.FullScreen,
-    container: Highcharts.HTMLDOMElement
-): void {
-    this.init(container.parentNode as any);
-} as any;
-
-FullScreen.prototype = {
-    /**
-     * Init function
-     * @private
-     * @param {Highcharts.HTMLDOMElement} container
-     *        Chart container's parent
-     * @return {void}
-     */
-    init: function (
-        this: Highcharts.FullScreen,
-        container: Highcharts.HTMLDOMElement
-    ): void {
-
-        let promise;
-
-        if (container.requestFullscreen) {
-            promise = container.requestFullscreen();
+H.Chart.prototype.toggleFullscreen = function (this: Highcharts.Chart): void {
+    var chart = this,
+        container = chart.container.parentNode as Highcharts.HTMLDOMElement;
```

Comment:
Hmm, I wonder why linter didn't catch vars...

Reply:
It looks like it is turned off for the migration phase. Still it would be good practice to use `let` at least. `let` and `const` get both transpiled, but it helps with the typing checks.